File: Blog.Web/ClientApp/src/app/core/models/Comment.ts
import { User } from './User';

/**
 * Comment model.
 */
export class Comment {
  /**
   * Post Comment Id
   * @param id number
   * Post Comment PostId
   * @param postId number
   * Post Comment AuthorId
   * @param userId number
   * Post Comment User
   * @param user User
   * Post Comment Email
   * @param email string
   * Post Comment string
   * @param name string
   * Post Comment commentBody
   * @param commentBody string
   * Post Comment Created At
   * @param createdAt Date
   */
  constructor(
    public id: number = 0,
    public postId?: number,
    public userId?: string,
    public user?: User,
    public email?: string,
    public name?: string,
    public commentBody: string = '',
    public createdAt?: Date
  ) {}
}

File: Data/Blog.Data.Models/Comment.cs
// <copyright file="Comment.cs" company="Blog">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>

namespace Blog.Data.Models
{
    using System;
    using Core;

    /// <summary>
    /// Comment model.
    /// </summary>
    /// <seealso cref="Entity" />
    public class Comment : Entity
    {
        /// <summary>
        /// Gets or sets the post identifier.
        /// </summary>
        /// <value>
        /// The post identifier.
        /// </value>
        public int PostId { get; set; }

        /// <summary>
        /// Gets or sets the post.
        /// </summary>
        /// <value>
        /// The post.
        /// </value>
        public virtual Post Post { get; set; }

        /// <summary>
        /// Gets or sets the comment body.
        /// </summary>
        /// <value>
        /// The comment body.
        /// </value>
        public string CommentBody { get; set; }

        /// <summary>
        /// Gets or sets the created at.
        /// </summary>
        /// <value>
        /// The created at.
        /// </value>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Gets or sets the likes.
        /// </summary>
        /// <value>
        /// The likes.
        /// </value>
        public int Likes { get; set; }

        /// <summary>
        /// Gets or sets the dislikes.
        /// </summary>
        /// <value>
        /// The dislikes.
        /// </value>
        public int Dislikes { get; set; }

        /// <summary>
        /// Gets or sets the user identifier.
        /// </summary>
        /// <value>
        /// The user identifier.
        /// </value>
        public string UserId { get; set; }

        /// <summary>
        /// Gets or sets the user.
        /// </summary>
        /// <value>
        /// The user.
        /// </value>
        public virtual ApplicationUser User { get; set; }

        /// <summary>
        /// Gets or sets the name.
        /// </summary>
        /// <value>
        /// The name.
        /// </value>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the email.
        /// </summary>
        /// <value>
        /// The email.
        /// </value>
        public string Email { get; set; }
    }
}